### ISSUE TITLE
Suggest when a function can be made const

### ISSUE THREAD

**Issue by JuanPotato** [rust-lang/rust-clippy#7184]
### What it does

Suggest when a function can be made a const fn

### Categories (optional)

- Kind: `clippy::perf`

*What is the advantage of the recommended code over the original code*

For example:
- Allows function to be called in a constant for free

### Drawbacks

None. Not sure why anyone wouldn't want a function to be const if it could be.

### Example

```rust
fn foo(a: i32, b: i32) -> i32 {
    a + b
}
```

Could be written as:

```rust
const fn foo(a: i32, b: i32) -> i32 {
    a + b
}
```

**Comment by mikerite** [comment]
We already have this lint: [missing_const_for_fn](https://rust-lang.github.io/rust-clippy/master/index.html#missing_const_for_fn) but it's allow by default. Maybe we need issue for getting it out of nursery.

**Comment by JuanPotato** [comment]
Oh my bad, it also looks like there are a few open issues regarding it
